timesketch/lib/analyzers/sequence_sessionizer.py

```diff
@@ -0,0 +1,158 @@
+"""Sequenced activity sessionizing sketch analyzer plugin."""
+
+from __future__ import unicode_literals
+
+from timesketch.lib.analyzers import sessionizer
+
+
+class SequenceSessionizerSketchPlugin(sessionizer.SessionizerSketchPlugin):
+    """Base class for specialized sequenced activity sessionizings
+    sketch plugins.
+
+    Attributes:
+        event_seq: List of dictionaries of attributes describing the events.
+        event_storage: List of events.
+        num_event_to_find: Number that shows which event from the sequence
+            should be matched.
+        recording: Shows if currently are finded and stored events that match
+            the specified sequence of events.
+        return_fields: List of name of event attributes, should be specified in
+            the inheriting sessionizers. It must contains 'timestamp'.
+        session_num: Counter for the number of sessions.
+        session_type: The name of the event sequence.
+    """
+    event_seq = None
+    event_storage = []
+    num_event_to_find = 0
+    recording = False
+    return_fields = ['timestamp']
+    session_num = 0
+    session_type = None
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Allocates each event between the first event of the event_seq and the
+        last event of the event_seq an session_type attribute and a session_num.
+
+        Returns:
+            String containing the name of the event sequence and the
+            number of sessions created.
+        """
+        if self.session_type is None:
+            raise RuntimeError('No session_type provided.')
+        if self.event_seq is None or self.event_seq == []:
+            raise RuntimeError('No event_seq provided.')
+        # If return_fields in none, then all attributes are provided.
+        if self.return_fields is not None:
```

Comment:
Didn't we discuss building a minimal list of `return_fields` depending on the dicts in `event_seq`? It's OK to leave it for a future PR but in this case please add a `# TODO` comment.

Reply:
My logic there is that if someone chosed to have all of the fields, he probably needs all of them.
If someone didn't specified the return_fields in his sessionizer, then return_fields remains ['timestamp'] and I am going to fill the rest of the fields that are needed. And a field is needed if event from event_seq has it.
If someone specified return_fields but not with the right attributes for some reason, I am not removing the not needed ones, I am just adding the attributes that are needed but missing. So in that case return_fields can end up with not used attributes.
But the another side of situation is if the person needs to access other attributes too, then to removing them is unexpected behaviour.
In the first situation I am adding extra complexity but making sure that the code will work as expected in the sequence_sessionizer. In the second situation, I can break someone's logic.

Reply:
What do you think it's better to be done?